### PR TITLE
[RNMobile] Disable Unsupported block editor for Reusable blocks

### PR DESCRIPTION
**Pull request by fluiddot**
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3067

## Description

Reusable blocks are no longer editable through UBE (Unsupported block editor) due to recent changes in the editing flow on the web version. Until we fully support editing them in the native editor, the edition through UBE should be disabled.

### `core/block` [Reusable block] _(only enabled in development mode)_
All UBE related logic has been removed, now on double tap on the block only a message is displayed.

### `missing/block` [Missing block]
A list with the unsupported blocks has been added to this block, this way we can control whether display the option for editing it using the web editor (UBE).

## How has this been tested?

Since these changes are related to unsupported blocks, it requires a quick setup to be done in the **web version editor**:
1. Create a post/page.
2. Add a block that is not supported in the native editor like GIF (`jetpack/gif`).
3. Create a Reusable block and add it.
4. Save the post/page.

The following flows have been used for testing these changes:

<details><summary>Reusable block (dev mode only)</summary>

**This has to be tested in development mode.**

### Steps
1. Go to a post/page that has a Reusable block.
2. Tap on it two times.
3. Check that there's no option to edit it through the web editor.

</details>

<details><summary>Missing block - UBE supported block</summary>

### Steps
1. Go to a post/page that has a block that is not supported in the native editor.
2. Tap on it two times.
3. Check that the option of editing it through the web editor is displayed.

</details>

<details><summary>Missing block - UBE unsupported block</summary>

**This has to be tested in release mode or by disabling the Reusable block from the block list in [`block-library` package](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/index.native.js#L228).** 

### Steps
1. Go to a post/page that has a Reusable block.
2. Tap on it two times.
3. Check that there's no option to edit it through the web editor.

</details>

## Screenshots 

<details><summary>Reusable block (dev mode only)</summary>

</details>

<details><summary>Missing block (jetpack/gif) - UBE supported block</summary>

</details>

<details><summary>Missing block (core/block) - UBE unsupported block</summary>

</details>

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by guarani — COMMENTED** [review]
Reviewed via https://github.com/wordpress-mobile/gutenberg-mobile/pull/3067#pullrequestreview-578692570

File: packages/block-library/src/missing/edit.native.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ import { applyFilters } from '@wordpress/hooks';
  */
 import styles from './style.scss';
 
+// Blocks that can't be edited through the Unsupported block editor identified by their name.
+const UBE_UNSUPPORTED_BLOCKS = [ 'core/block' ];
```
[the remark this answers is not in the document]
Comment (fluiddot) [reply packages/block-library/src/missing/edit.native.js]: I agree, looks confusing to use the word `unsupported` combined with `Unsupported block editor` 😄 , I've just [renamed](https://github.com/WordPress/gutenberg/pull/28552/commits/5d3e1dee73bea4b234044a0cb1ae0b88bc3f802a) it to `UBE_INCOMPATIBLE_BLOCKS`.

_PS: If a block is not supported by the Unsupported block editor, wouldn't that mean that is a supported block? 🤣_

Thanks @guarani for the review!

[the remark this answers is not in the document]
Comment (guarani) [reply packages/block-library/src/missing/edit.native.js]: > PS: If a block is not supported by the Unsupported block editor, wouldn't that mean that is a supported block? 🤣

🤣

**Review by guarani — APPROVED** [review]
Tested via https://github.com/wordpress-mobile/gutenberg-mobile/pull/3067#pullrequestreview-579332648.

LGTM!

**Comment by fluiddot** [comment]
I've tried to re-run the PR checks multiple times but they keep failing, I don't think they're failing due to these PR changes but I'm concerned about merging it with some checks in red.

However I saw that most of the recent merged PRs had the same issue so maybe it's safe to merge it 🤔 .

**Comment by fluiddot** [comment]
> I've tried to re-run the PR checks multiple times but they keep failing, I don't think they're failing due to these PR changes but I'm concerned about merging it with some checks in red.
> 
> However I saw that most of the recent merged PRs had the same issue so maybe it's safe to merge it 🤔 .

On `master` branch the same checks are failing so looks like it's not really related to the PR changes, therefore I'll merge it.

- [End-to-End Tests / Admin - 1 (pull_request)](https://github.com/WordPress/gutenberg/runs/1805172581)
- [End-to-End Tests / Admin - 3 (pull_request)](https://github.com/WordPress/gutenberg/runs/1805172623)
- [End-to-End Tests / Admin - 4 (pull_request)](https://github.com/WordPress/gutenberg/runs/1805172646)
- [Unit Tests / PHP (pull_request)](https://github.com/WordPress/gutenberg/runs/1805172652)

**Pull request merged**